### DISCUSSION TITLE
Skipping uploading for quota "exceeded"

### DISCUSSION THREAD

**Discussion by brakkio86** [abraunegg/onedrive#3039] [Q&A] [answered]
Hello,

I'm recently experimenting this type of error:

```
[...]
Skipping uploading this new file as it exceeds the available free space on Microsoft OneDrive: ./foo.txt
[...]
```

The problem is similar to issue here https://github.com/abraunegg/onedrive/issues/3005

From debugging I get this answer from API:

`DEBUG: Quota Details: {"@odata.context":"https:\/\/graph.microsoft.com\/v1.0\/$metadata#drives\/$entity","quota":{"deleted":312817474,"remaining":0,"state":"exceeded","total":3298534883328,"used":3750732836680}}`

that, reformatting, is:

```json
{
  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#drives/$entity",
  "quota": {
    "deleted": 312817474,
    "remaining": 0,
    "state": "exceeded",
    "total": 3298534883328,
    "used": 3750732836680
  }
}
```

On the other hand, on web-ui I see about 900 GiB available. Moreover, original onedrive client (on windows), via web-ui and `rclone` tool can upload without any issue.

Could be same misbehaviour about microsoft api that cause this issue? Is it possible to disable quota check on onedrive client?

Thanks,
Francesco

**Comment by abraunegg** [comment, marked as answer]
@brakkio86 
> Could be same misbehaviour about microsoft api that cause this issue? Is it possible to disable quota check on onedrive client?

You have exceeded your quota - this has nothing to do with this client, but your data on Microsoft OneDrive. The API is responding that you have exceeded your quota.

This client adheres to this - that the WebUI and/or rclone ignore this, I cannot advise why this is being ignored. 

This is not ignored by this client to ensure that you have data integrity - the last thing I want to do is have you upload a file, *think* you have uploaded it, but it is somehow silently rejected by Microsoft OneDrive and you experience data loss.

Your choices are:

1. Clean up your online space
2. Delete unused space
3. Purchase more space

There is nothing here for me to do / change - sorry.

And to be clear - I will *not* implement a bypass for ignoring the Microsoft OneDrive API stating you have exceeded your space online.

**Comment by brakkio86** [comment]
I'm sorry, but I don't agree with this claim _"You have exceeded your quota"_ but I understand that this client just read the API as it is. I think there is some explanation of what microsoft api means.

I've checked with other client and the file are uploaded correctly to four type of devices:

1. windows pc with original onedrive client
2. android phone with an app to upload files
3. linux "NAS" that download all drive (for backup) with rclone
4. laptop with fedora and this client

Only the fourth is having problem to upload files and the easiest answer normally is the correct one. Therefore there problem is only on the fourth case, on this client.

Right now,  I'm workaround this issue uploading every files with rclone, then a manual sync of this client for database consistency, and ends ok (I've uploaded files before). I've tried a full resync, but the same issue.

I'm not saying that is a bug, but maybe same misbehaviour or misunderstanding about API response.

Right now, I'm going to downgrade to 2.4.X, full resync and check if gives me the same behaviour.

Thanks,
Francesco

**Comment by abraunegg** [comment]
@brakkio86 
> I'm not saying that is a bug, but maybe same misbehaviour or misunderstanding about API response.

This is *not* a bug. 

Please read:
* https://support.microsoft.com/en-us/office/microsoft-storage-quotas-8f2f9d72-04d1-4223-a5ae-c2fdd26dd770

![image](https://github.com/user-attachments/assets/e4c1c07e-6e74-4e97-9129-edc455962320)



The Microsoft API is responding with this data .... the client is not *generating* this data.

**Reply by brakkio86** [reply]
Thanks for the answer, but the link is related to OneDrive Basic or Family, I'm using OneDrive Business and internal metric says that I have 963,57 GB available. I'm not saying that is a **bug** of this client as I've not opened an issue, but open a discussion to understand my issue using this client compared to the original one and two other open source implementation that does not manifest any quota warning.

Right now, I'm trying to go directly to the original API documentation [here](https://learn.microsoft.com/en-us/graph/) but I'm not comfortable to read Microsoft documentation. Maybe there is someone that can help to understand microsoft api.

**Reply by abraunegg** [reply]
As you have OneDrive Business account, you have Microsoft Support.

Please raise a request with Microsoft and ask them why their API is responding as 'exceeded'. 

I do not work for Microsoft - I cannot assist you further.
